### PR TITLE
Fix engine path resolution

### PR DESCRIPTION
**Pull request by pcoliveira**
Closes issue #56 
BREAKING CHANGE: The returned path for mounted engines now includes the path prefix.

An error was raised when testing a route that is mounted after an engine that shares the same route prefix.
The path returned for a mounted engine didn't include the route prefix (only the engine sub path was returned).

On the engine test case, the expected path was changed from `/eng_route` to `/my_engine/eng_route`, to match the mounted path.

### CODE REVIEW

**Comment by xcskier56** [comment]
@pcoliveira is there anything I can do to help this PR along? This has been nagging me for a long while and finally has become more of an issue

File: lib/rspec/openapi/record_builder.rb [hunk 1]
```diff
@@ -99,21 +107,18 @@ def rack_test?(context)
...
-        return route
+        if route.app.engine?
+          route, path = find_rails_route(request, app: route.app.app, path_prefix: path)
+          next if route.nil?
+        end
+        return [route, path_prefix + path]
       end
     end
-    raise "No route matched for #{request.request_method} #{request.path_info}"
+    nil
```
Comment (exoego) [inline lib/rspec/openapi/record_builder.rb:121]: Q) Why this error message is removed?

**Comment by argent-codes** [comment]
My organization recently ran into the issue fixed by this PR, so I would love to see this released ❤️

**Comment by exoego** [comment]
@pcoliveira Hi. Sorry for late response. 
Could you check your PR?
I left a question.

File: lib/rspec/openapi/record_builder.rb (hunk 1 above)
Comment (pcoliveira) [reply lib/rspec/openapi/record_builder.rb:121]: When two engines are mounted on the same root path, and the first doesn't resolve, the resolution needs to continue to the next engine. The raise was moved to the caller, when all engines are resolved

**Comment by pcoliveira** [comment]
> @pcoliveira Hi. Sorry for late response. Could you check your PR? I left a question.

sorry, I missed your question.

**Comment by pcoliveira** [comment]
@exoego How may I help? is there anything I need to do to move the PR forward?

**Review by exoego — APPROVED** [review]
LGTM
This is amazing.
Sorry for the late response.

**Pull request merged**